HwProj.CoursesService/HwProj.CoursesService.API/Services/CoursesService.cs

```diff
@@ -306,6 +306,15 @@ public async Task<bool> HasStudent(long courseId, string studentId)
             return student != null;
         }
 
+        public async Task<bool> IsCourseUser(long courseId, string userId)
+        {
+            var course = await _coursesRepository.GetAsync(courseId);
+            var isCourseMentor = course.MentorIds.Contains(userId);
+            var isCourseStudent = await HasStudent(courseId, userId);
+
+            return course.IsOpen || isCourseMentor || isCourseStudent;
```

Comment:
Давай оптимизируем для открытых курсов
```suggestion
            var course = await _coursesRepository.GetAsync(courseId);
            if (course.IsOpen) return true;
            
            var isCourseMentor = course.MentorIds.Contains(userId);
            var isCourseStudent = course.CourseMates.Any(...);
            return isCourseMentor || isCourseStudent;
```